File: Python/539. Minimum Time Difference.py
class Solution:
    def findMinDifference(self, timePoints):
        """
        :type timePoints: List[str]
        :rtype: int
        """

        for i in range(len(timePoints)):  # Turn into minutes from 0 to this time
            timePoints[i] = int(timePoints[i][0:2])*60 + int(timePoints[i][3:])

        timePoints = sorted(timePoints)  # Time sorted from small to large

        # Time difference before and after 0 o'clock
        ans = 1440 - timePoints[-1] + timePoints[0]

        for i in range(1, len(timePoints)):
            # Adjacent time difference
            ans = min(ans, timePoints[i] - timePoints[i-1])

        return ans

File: C++/539. Minimum Time Difference.cpp
class Solution {
public:
    static bool cmp1(int a, int b) {
        return a < b;
    }
    int findMinDifference(vector<string>& timePoints) {
        vector<int> time_points;
        for (auto &it : timePoints) {
            int temp_time = ((it[0] - '0') * 10 + (it[1] - '0') ) * 60 + ((it[3] - '0') * 10 + (it[4] - '0'));
            time_points.push_back(temp_time);
        }// convert string_time to int_time

        sort(time_points.begin(), time_points.end(), cmp1); // sort int_time from small to large
        int ans = 999999;

        for (int i = 1; i < time_points.size(); i++) {
            ans = min(time_points[i] - time_points[i - 1], ans);
        }// find minimun minutes between Adjacent time points
        ans = min(ans, 1440 - time_points[time_points.size() - 1] + time_points[0]); // situation last point and first point
        return ans;
    }
};